Functional into main

Enonome opened the pull request:
Pull request to add the Functional document into the main branch

Enonome committed:
Create functional_specifications.md

Starting the Functional Spec

Enonome committed:
Merge branch 'main' into functional

Enonome committed:
Hierarchy documents

Enonome committed:
Update functional_specifications.md

Update the Functional template

Enonome committed:
Update functional_specifications.md

Updating the table of contents, beginning of the #1 (1.1, 1.2, 1.3, 1.4). Also, updating/changing the description of some of the title

Enonome committed:
Updating functional_specifications.md

Adding the #2 section, with 2 images related to the Dijkstra algorithm
(Done on 01/14)

Enonome committed:
Updating the functional_specifications.md

Adding the #3 section (including the 3.1, 3.2 and 3.3) and an image of our logo

Enonome committed:
Update functional_specifications.md

Updating the 3.2 section; adding the Functional requirements (6th section) and a MoSCoW table (9th section).
Removing errors in some sections

Enonome committed:
Update functional_specifications.md

Changing some info in the 2.1 and C (2.2) sections;
Adding the MoSCoW table

Enonome committed:
Update functional_specifications.md

Updating the name of the redirections into a few sections.
Also, filling the 4th and 5th section.
Finally, starting the 6th section

Enonome committed:
Update functional_specifications.md

Merging the 2 table of the MoSCoW table

Enonome committed:
Update functional_specifications.md

Changing some info in the 3.1th section.
Adding MacOS specifications
Removing comments
Resolving an error with the html table style

Commit:
Update functional_specification

> Adding personas
> Adapting images name to the technical format

Enonome committed:
Update functional_specifications.md

Formatting the personas table and specification table
Adding the 6.1 section

Commit:
Update functional_specification.md

> Fixing personas

Enonome committed:
Update functional_specification.md

Formatting the personas table (like the previous comment, error during the commit)
Adding C section (in 6.1 section)

Enonome committed:
Remove images

Removing the images because of a problem caused with the name of them

Enonome committed:
Re-adding the images

Fixing the issue (names)

Enonome committed:
Update functional_specifications.md

Adding "br" tags (no line break before)
Removing some comments
Filling the 7th section

Enonome committed:
Update functional_specifications.md

Pre-final version (pending for a review)

[- Adding the 1.5 section
 - Formatting the tables
 - Correcting typos
 - Filling the Glossary and the 8th, 10th sections]

Enonome committed:
Update functional_specification.md

Trying to resolve issues with the GIF (and formatting the second image and its text)

Enonome committed:
Update functional_specifications.md

Pre-final version +

Enonome committed:
Update functional_specifications.md

(Changing some name sections)

Enonome committed:
Update functional_specifications.md

Changing some name sections (2nd version)

Petite-Brioche0 commented:
- file name : "functional_specifications.md" -> "functional_specification.md"
- lines 3, 20, 59, 76, 118,  : "Functional **Specifications**" -> "Functional **Specification**"
- line 59 : "You can found **bellow**" -> "**below**"
- lines 63 to 68 : Written "**Github** Profile" rather than "**LinkedIn** Profile"
- line 65 : "the language will be **use**” -> "**used**"
- line 72 : "“We **decide** to name" -> "**decided**"
- line 72 : "a **played of** words" -> "a **play on** words"
- line 76 : "the features we **decide** to skip" -> "**decided**"
- line 77 : “please **refer you** to the" -> "please **refer** to the"
- line 82 : "William the **project manager**" -> "**Project Manager**"
- line 123 : "31**th**" -> "31**st**"
- line 131 : "it will **be ask** you" -> "it will **ask** you"
- line 134 : “### 2.2 What the application used to work?” -> "### 2.2 What does the application use to work?” (don't forget to remake the table of content)
- line 148 : "needed to be **implement** by" -> "**implemented**"
- line 152 : "how we have **turn** the" -> "**turned**"
- line 156 : “The software has to **respects** criteria" -> "**respect**"
- line 158 : "enter your **depart** point” -> "**departure**"
- line 167 : "**precess**" -> "**process**"
- line 186 : “Once **is** done" -> "Once **it is** done"
- line 204 : "no mobile platforms is **indicate**" -> "**indicated**"
- line 208 : "**Fell** free to go" -> "**Feel**"
- line 276 : "process **make**" -> "**makes**"
- line 279 : "the time between **to** points" -> "**two**"
- line 292 : “The API[^7] **have** to respond" -> "**has**"
- line 300 : "by the user **have** to be correct" -> "**has**"
- line 308 : "can be easily **test**" -> "**tested**"
- line 357 : "A loading bar that **displayed**" -> "**displays**"
- line 375 : “the user **want** to utilize” → “**wants**”
- lines 349 to 361 : the words reference doesn't work

Mattstar64 commented:
reopening PR

Enonome committed:
Update functional_specification.md

Document modification following QA review.

Enonome committed:
Update functional_specification.md

Deleting the sources for the Glossary into the MoSCoW table

Petite-Brioche0 commented on documents/functional/functional_specification.md:
```diff
@@ -56,30 +56,30 @@
 ## 1.Introduction
 
-This document is the functional specifications of the 3rd Project of the year called Quickest Path. You can found bellow the name and the role of each member of the team:
+This document is the functional specification of the 3rd Project of the year called Quickest Path. You can found below the name and the role of each member of the team:
```
found -> find

Enonome replied:
Corrected

Petite-Brioche0 commented on documents/functional/functional_specification.md:
```diff
@@ -56,30 +56,30 @@
 ### 1.1 Project Overview
 
-The aim of the project is to create a software that find the quickest path between two landmarks in the United States. To achieve this request, we have to use the C++ language[^1]. We decide to name our software "QPS", aka "Quickest Path System", a played of words among GPS "Global Positioning System" and the name of the project "Quickest Path".
+The aim of the project is to create a software that find the quickest path between two landmarks in the United States. To achieve this request, we have to use the C++ language[^1]. We decided to name our software "QPS", aka "Quickest Path System", a play on words among GPS "Global Positioning System" and the name of the project "Quickest Path".
```
find -> finds

Enonome replied:
Done

Petite-Brioche0 commented on documents/functional/functional_specification.md:
```diff
@@ -145,17 +145,17 @@ The final result looks like this --> <img src="../images/functional/final_dijkst
 ### 3.1 Performance
 
-The software has to respects criteria to be competitive and follows ours objectives. Once you've initialized the verification process (including the file with landmarks), the command prompt must finish the checking, following the number of lines there are in your file (1 minute for 30,000,000 lines), so depending on the file size. <br>
+The software has to respect criteria to be competitive and follows ours objectives. Once you've initialized the verification process (including the file with landmarks), the command prompt must finish the checking, following the number of lines there are in your file (1 minute for 30,000,000 lines), so depending on the file size. <br>
```
ours -> our

Enonome replied:
Corrected

Petite-Brioche0 commented on documents/functional/functional_specification.md:
```diff
@@ -164,7 +164,7 @@ Finally, the software has to indicate the most optimized path/the quickest path,
 
 #### 3.2.2 Verification
-While you're waiting for the verification precess, a percentage bar will be displayed to don't leave you in the dark. After 10 seconds, if the bar didn't progress of any percent, please restart the verification.
+While you're waiting for the verification process, a percentage bar will be displayed to don't leave you in the dark. After 10 seconds, if the bar didn't progress of any percent, please restart the verification.
```
to don't leave you in the dark -> to avoid leaving you in the dark

Enonome replied:
Done

Petite-Brioche0 reviewed (changes requested):
added 4 news changement before merging

Enonome committed:
Update functional_specification.md

Correcting the last typos
Modifying code blocks to match technical specifications

Petite-Brioche0 reviewed (approved):
Everything seems good

GuillaumeDespaux reviewed (changes requested):
1. Table of contents
- [x] Change all the title with the same convention (This Is A Title)

Enonome committed:
Update functional_specification.md

Changing title names (conventions must be the same for all titles)

GuillaumeDespaux reviewed (approved):
Ok for me.
